### PR TITLE
feat: appkit demo

### PR DESCRIPTION
**Pull request by enesozturk**
# Description

Introducing AppKit demo application. This PR includes changes to have the initial changes for Demo (aka previously Builder) application.

Main features: 
- Customize connection methods
- Customize wallet features
- Customize theme
- Share customization via URL

All the required changes to AppKit has been handled separate PRs. See:
- https://github.com/reown-com/appkit/pull/3412
- https://github.com/reown-com/appkit/pull/3400
- https://github.com/reown-com/appkit/pull/3407

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-1424, APKT-1426, APKT-650, APKT-649, APKT-646, APKT-1466, APKT-1465, APKT-775, APKT-1585, APKT-1584, APKT-1681

# Showcase (Optional)

https://github.com/user-attachments/assets/7768776f-d656-490f-ba1a-4a5feac5646d

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [ ] I have filled out all required sections
- [x] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

**Review by magiziz — APPROVED** [review]

File: apps/builder/app/page.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use client'
+
+import { useAppKitContext } from '@/hooks/use-appkit'
+import { SidebarContent } from '@/components/sidebar-content'
+import { PreviewContent } from '@/components/preview-content'
+import { cn } from '@/lib/utils'
+import { BrandingHeader } from '@/components/branding-header'
+
+export default function Page() {
+  const { config } = useAppKitContext()
```
Comment (tomiir) [inline apps/builder/app/page.tsx:10]: are you using this somewhere?

File: apps/builder/components/configuration-sections/section-wallet-features.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+export function SectionWalletFeatures() {
+  const { config, updateFeatures } = useAppKitContext()
+  const connectMethodsOrder = config.features.walletFeaturesOrder || defaultWalletFeaturesOrder
+
+  function handleNewOrder(items: UniqueIdentifier[]) {
+    const titleValueMap = {
+      Buy: 'onramp',
+      Swap: 'swaps',
+      Receive: 'receive',
```
Comment (zoruka) [inline apps/builder/components/configuration-sections/section-wallet-features.tsx:24]: Does `Receive` feature changes anything in the Modal? I could not see differences in the preview

**Review by zoruka — APPROVED** [review]
The preview is looking really good 🚀

File: apps/builder/components/configuration-sections/section-wallet-features.tsx (hunk 2 above)
Comment (enesozturk) [reply apps/builder/components/configuration-sections/section-wallet-features.tsx:24]: It's only visible on the embedded wallet connection. The default (old) account view doesn't have this as one of the options in the list

**Review by svenvoskamp — APPROVED** [review]
Amazing work ✨ 

Left 1 small comment

**Pull request merged**